plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitManager.java

```diff
@@ -108,6 +112,36 @@ public ConnectorSplitSource getSplits(
         return new ClassLoaderSafeConnectorSplitSource(splitSource, IcebergSplitManager.class.getClassLoader());
     }
 
+    private Scan<?, FileScanTask, CombinedScanTask> getScan(IcebergMetadata icebergMetadata, Table icebergTable, IcebergTableHandle table, ExecutorService executor)
+    {
+        Long fromSnapshot = icebergMetadata.getIncrementalRefreshFromSnapshot(table.getSchemaTableName()).orElse(null);
+        if (fromSnapshot != null) {
+            // check if fromSnapshot is still part of the table's snapshot history
+            boolean containsFromSnapshot = SnapshotUtil.isAncestorOf(icebergTable, fromSnapshot);
```

Comment:
does compation re-enable incremental refreshes?

Reply:
Unfortunately not, because the updated/deleted rows will still be present in the MV, so it wouldn't be enough to just append the new/up-to-date rows into the MV, but we would also need to ensure that the obsolete records are purged from the MV. We can probably make this work though with some extra work in the subsequent phases, but that probably would require plan recompilation.